[ashleyy2444] iP

ashleyy2444 opened the pull request:
# YourManager
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

*YourManager* frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn :+1:
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. clone the repository using `git clone` ([link](https://github.com/ashleyy2444/ip))
2. add your tasks
3. let it manage your tasks for you :smile:

And it is **FREE**!

## Features:

- [X] Managing tasks
- [X] Managing deadlines 
- [X] Managing events
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method::
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ashleyy2444 committed:
Rename, Greet, Exit

ashleyy2444 committed:
Echo

ashleyy2444 committed:
Add, List

ashleyy2444 committed:
Mark as Done

ashleyy2444 committed:
ToDos, Events, Deadlines

ashleyy2444 committed:
Automated Text UI Testing

ashleyy2444 committed:
Handle Errors

ashleyy2444 committed:
Delete

ashleyy2444 committed:
Updating OOP - 1

ashleyy2444 committed:
Update OOP - 2

ashleyy2444 committed:
Update OOP - 3

ashleyy2444 committed:
Implement Level 7

ashleyy2444 committed:
Save

ashleyy2444 committed:
Implement Level 8

ashleyy2444 committed:
Implement Level 8

ashleyy2444 committed:
Merge pull request #1 from ashleyy2444/branch-Level-8

Implement Level 8

ashleyy2444 committed:
Merge branch 'branch-Level-8'

ashleyy2444 committed:
Merge remote-tracking branch 'origin/master'

hjungwoo01 commented on src/main/java/CommandHandler.java:
```diff
@@ -0,0 +1,115 @@
+import java.util.ArrayList;
+
+public class CommandHandler {
+    public static void handleCommand(String userInput, ArrayList<Task> tasks, int taskCounter) throws DukeException {
```
I think it would be better to use List<Task> instead of ArrayList<Task> in function headers, as List is the parent class, which adheres to good software engineering principles.

hjungwoo01 commented on src/main/java/CommandHandler.java:
```diff
@@ -0,0 +1,115 @@
+        Task removedTask = tasks.remove(index);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(removedTask.getStatusIcon());
+        System.out.println("Now you have " + tasks.size() + " " + (tasks.size() <= 1 ? "task" : "tasks") + " in the list.");
```
The following line of code is quite inline, which may affect readability. It combines multiple operations in a single statement. Consider breaking it down into smaller, more understandable parts for improved clarity.

hjungwoo01 reviewed (commented):
Overall, I found your code to be quite well-written. It was generally easy to read, and I didn't notice any major issues. While there were a few places with deep nesting, they were relatively minor and didn't significantly impact overall readability. Additionally, I didn't really notice any coding standard violations. Great job overall!

SherwynNg commented on src/main/java/CommandHandler.java:
```diff
@@ -0,0 +1,115 @@
+        tasks.get(index).markAsDone();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(tasks.get(index).getStatusIcon());
+    }
```
Perhaps can consider creating a UI class to handle all the prints.

SherwynNg commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,44 @@
+        taskCounter++;
+        System.out.println("Got it. I've added this task:");
+        System.out.println(tasks.get(tasks.size() - 1).getStatusIcon());
+        System.out.println("Now you have " + tasks.size() + " task" + (tasks.size() == 1 ? "" : "s") + " in the list.");
```
Can consider placing a line break to improve readability.

SherwynNg commented on src/main/java/TaskReader.java:
```diff
@@ -0,0 +1,71 @@
+        }
+    }
+
+    private static LocalDate parseDate(String dateString) {
```
Great use of verbs and camel case for naming methods.

SherwynNg reviewed (commented):
Overall, the code is very readable 👍 .
Might want to consider breaking longer lines, to improve readability. Other than that the great practice of OOP principles and coding standards.

ashleyy2444 committed:
Extract Ui and TaskList Class

ashleyy2444 committed:
Extract Storage Class

ashleyy2444 committed:
Extract Parser Class

ashleyy2444 committed:
Group classes into 'duke' package

ashleyy2444 committed:
Merge branch 'add-gradle-support'

ashleyy2444 committed:
Set up gradle

ashleyy2444 committed:
Add JUnit tests for Todo and Deadline

ashleyy2444 committed:
Package app as a Jar file

ashleyy2444 committed:
Add JavaDoc comments

ashleyy2444 committed:
Update README.md

ashleyy2444 committed:
Update README.md

ashleyy2444 committed:
Update README.md

ashleyy2444 committed:
Update the code to comply with coding standard

Chiarena commented on src/main/java/duke/CommandHandler.java:
```diff
@@ -0,0 +1,119 @@
+        String[] parsedCommand = Parser.parse(userInput);
+        String commandType = parsedCommand[0].toLowerCase();
+
+        switch (commandType) {
```
i liked the naming of your different handle functions, they were very clear and easy to understand

Chiarena commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @Override
+    public String getStatusIcon() {
```
Is there a clearer way to name the getStatusIcon method?

Chiarena commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,48 @@
+        return userInput.split(" ", 2);
+    }
+
+    public static LocalDate parseDate(String dateString) throws DukeException {
```
i like how you named your parse methods

Chiarena commented on src/main/java/duke/TaskReader.java:
```diff
@@ -0,0 +1,73 @@
+        }
+    }
+
+    private static Task createTaskFromLine(String line) {
```
what did you mean by create?

Chiarena reviewed (commented):
Overall very clear naming conventions and organisation of code that made it easy to read and understand

ashleyy2444 committed:
Add function of using keyword to find task

ashleyy2444 committed:
Merge branches 'branch-A-JavaDoc' and 'branch-Level-9'

ashleyy2444 committed:
Merge remote-tracking branch 'origin/master'

ashleyy2444 committed:
Add GUI to chatbot

ashleyy2444 committed:
Delete unused files

ashleyy2444 committed:
Fix errors with GUI

ashleyy2444 committed:
Merge branch 'branch-Level-10'

ashleyy2444 committed:
Add assertions to commands

Assertions are enabled in build.gradle. The codebase lacks assertions, making it challenging to ensure the expected behaviour of various methods.

Assertions are added in methods related to the different commands to ensure that specific conditions are met during runtime and to check if the user input is handled appropriately.

Assert statements have been added to the methods addToDoTask, addDeadlineTask, addEventTask, deleteTask, markTaskAsDone, markTaskAsNotDone, and findTasks which are used in the CommandHandler class. These assertions verify specific conditions, such as valid input lengths and successful task additions or modifications.

This will help to check assumptions and help catch potential issues during development.

ashleyy2444 committed:
Merge pull request #3 from ashleyy2444/branch-A-Assertions

Add assertions to commands

ashleyy2444 committed:
Improve code quality

The code contains unnecessary binary files (*.class).

Tracking binary files in version control can lead to unnecessary commits, making it harder to review and understand changes. It also creates more significant overhead during branching, merging, and cloning.

Remove unnecessary binary files from version control, ensuring a cleaner repository.

Binary files, such as compiled Java classes, don't belong in version control. They should be generated locally during the build process. Using `.gitignore` prevents accidentally committing them.

ashleyy2444 committed:
Merge pull request #4 from ashleyy2444/branch-A-CodeQuality

Improve code quality

ashleyy2444 committed:
Generate JAR file

ashleyy2444 committed:
Rename Duke class to Georgie class

ashleyy2444 committed:
Add date for Event class

ashleyy2444 committed:
Rename DukeException class to GeorgieException class

ashleyy2444 committed:
Rename package from duke to georgie

ashleyy2444 committed:
Deal with duplicate items

ashleyy2444 committed:
Resolve errors for handling deadlines

ashleyy2444 committed:
Resolve errors for handling events

ashleyy2444 committed:
Resolve errors for dealing with duplicates

ashleyy2444 committed:
Ensure name of product is 'Georgie'

ashleyy2444 committed:
Add user guide

ashleyy2444 committed:
Update user guide

ashleyy2444 committed:
Improve code quality

ashleyy2444 committed:
Remove height limit for Dialog Box

ashleyy2444 committed:
Ensure file can be loaded and saved

ashleyy2444 committed:
Handle parsing errors

ashleyy2444 committed:
Handle mark and unmark in georgie.txt

ashleyy2444 committed:
Handle deadline parsing error

ashleyy2444 committed:
Remove irrelevant imports
